src/mon/OSDMonitor.cc

```diff
@@ -7111,6 +7111,26 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     ss << osdmap.get_crush_version() + 1;
     goto update;
 
+  } else if (prefix == "osd crush set-all-straw-buckets-to-straw2") {
+    CrushWrapper newcrush;
+    _get_pending_crush(newcrush);
+    for (int b = 0; b < newcrush.get_max_buckets(); ++b) {
```

Comment:
@liewegas what i mean is something like this:

```c++
unsigned converted = 0;
for (...) {
  if (...) {
    // ...
    newcrush.bucket_set_alg(bid, CRUSH_BUCKET_STRAW2);
    converted++;
  }
}
if (!converted) {
  err = 0;
  ss << "no rules converted";
  goto reply;
}
```

Reply:
Yeah, but we might not be converting anything because the unstable in-flight crush map doesn't need a change because we are racing with someone else making the same update.  In that case we still have to wait for paxos to commit.

Reply:
oic! because the in-flight change **could** also contain the expected change of this command. so to be on the safe side, we need to wait.

thanks for your patience!